skyportal/tests/frontend/test_followup_requests.py

```diff
@@ -0,0 +1,201 @@
+import uuid
+import pytest
+import requests
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.common.action_chains import ActionChains
+from baselayer.app.env import load_env
+from skyportal.tests import api
+
+env, cfg = load_env()
+endpoint = cfg['app.sedm_endpoint']
+sedm_isonline = requests.get(endpoint, timeout=5).status_code in [200, 400]
+
+
+def add_telescope_and_instrument(instrument_name, token):
+    status, data = api("GET", f"instrument?name={instrument_name}", token=token)
+    if len(data["data"]) == 1:
+        return data["data"][0]
+
+    telescope_name = str(uuid.uuid4())
+    status, data = api(
+        "POST",
+        "telescope",
+        data={
+            "name": telescope_name,
+            "nickname": telescope_name,
+            "lat": 0.0,
+            "lon": 0.0,
+            "elevation": 0.0,
+            "diameter": 10.0,
+            "robotic": True,
+        },
+        token=token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+    telescope_id = data["data"]["id"]
+
+    status, data = api(
+        "POST",
+        "instrument",
+        data={
+            "name": instrument_name,
+            "type": "imager",
+            "band": "Optical",
+            "telescope_id": telescope_id,
+            "filters": ["ztfg"],
+            "api_classname": "SEDMAPI",
+        },
+        token=token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+    return data["data"]
+
+
+def add_allocation(instrument_id, group_id, token):
+    status, data = api(
+        "POST",
+        "allocation",
+        data={
+            "group_id": group_id,
+            "instrument_id": instrument_id,
+            "hours_allocated": 100,
+            "pi": "Ed Hubble",
+        },
+        token=token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+    return data["data"]
+
+
+def _followup_request_test_body(
```

Comment:
Should this be a fixture?

Reply:
Thought about it for a while and I think it might be better to keep as a method in `test_followup_requests` for now as it not only creates and returns data and state to be used in tests (like most other fixtures) but it also runs test statements themselves (assert, wait for xpath, etc.)

Reply:
What do you think? I renamed it, for clarity, to `add_followup_request_using_frontend_and_verify`.